pennylane/beta/tapes/qnode.py

```diff
@@ -383,11 +383,38 @@ def construct(self, args, kwargs):
             )
 
     def __call__(self, *args, **kwargs):
+
+        if self.interface == "autograd":
+            # HOTFIX: to maintain compatibility with core, here we treat
+            # all inputs that do not explicitly specify `requires_grad=False`
+            # as trainable. This should be removed at some point, forcing users
+            # to specify `requires_grad=True` for trainable parameters.
+            args = [
+                anp.array(a, requires_grad=True) if not hasattr(a, "requires_grad") else a
+                for a in args
+            ]
+
         # construct the tape
         self.construct(args, kwargs)
 
         # execute the tape
-        return self.qtape.execute(device=self.device)
+        res = self.qtape.execute(device=self.device)
+
+        if list(res.shape) == [1]:
+            # HOTFIX: to maintain compatibility with core, we squeeze
+            # any output that has shape (1,).
+
+            # Get the namespace associated with the return type
+            res_type_namespace = res.__class__.__module__.split(".")[0]

```

Comment:
Is this idiot proof? For example when the return type is in a sub module? (But I am not sure what's going on here, so ignore if so)

Reply:
Here is what it looks like for the (known) datatypes we will have:

```python
In [48]: np.array([0.1]).__class__.__module__
Out[48]: 'numpy'

In [49]: torch.tensor([0.1]).__class__.__module__
Out[49]: 'torch'

In [50]: tf.constant([0.1]).__class__.__module__
Out[50]: 'tensorflow.python.framework.ops'

In [51]: qml.numpy.array([0.1]).__class__.__module__
Out[51]: 'pennylane.numpy.tensor'
```

The `.split(".")` drops all submodules

Reply:
This is definitely a bit of a 'hack', but unfortunately we can't rely on interfaces, since the underlying device might be a 'backprop' device that natively returns a specific datatype